### PR TITLE
🌱 Add codecov

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Add codecov to keep an overview over our test coverage

(mostly copied from CAPZ)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2308

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold
Still trying it out locally, playing a bit around with it

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+coverage:
```
Comment (sbueringer) [inline .github/codecov.yml:1]: Documentation: https://docs.codecov.com/docs/commit-status

Comment (sbueringer) [reply .github/codecov.yml:1]: I think 1% is fine given we use informational

File: .github/workflows/cover.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
```
Comment (killianmuldoon) [inline .github/workflows/cover.yaml:1]: Do we need this?

Comment (sbueringer) [reply .github/workflows/cover.yaml:1]: Thx, copy paste..

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

Happy to merge this and see how happy we are with the noise over time.

**Pull request merged**